apiml-security-common/src/main/java/org/zowe/apiml/security/common/verify/TrustedCertificatesProvider.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ */
+
+package org.zowe.apiml.security.common.verify;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.http.HttpResponse;
+import org.apache.http.HttpStatus;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.util.EntityUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Service;
+import org.zowe.apiml.message.log.ApimlLogger;
+import org.zowe.apiml.product.logging.annotations.InjectApimlLogger;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.charset.StandardCharsets;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateException;
+import java.security.cert.CertificateFactory;
+import java.util.*;
+
+@Service
+@Slf4j
+public class TrustedCertificatesProvider {
+
+    private final Set<String> publicKeyCertificatesBase64;
+    private final CloseableHttpClient httpClient;
+
+    @InjectApimlLogger
+    private final ApimlLogger apimlLog = ApimlLogger.empty();
+
+    @Autowired
+    public TrustedCertificatesProvider(@Qualifier("secureHttpClientWithoutKeystore") CloseableHttpClient httpClient,
+                                       @Qualifier("publicKeyCertificatesBase64") Set<String> publicKeyCertificatesBase64) {
+        this.httpClient = httpClient;
+        this.publicKeyCertificatesBase64 = publicKeyCertificatesBase64;
+    }
+
+    //    @Cacheable(value = "certificates", key = "#certificatesEndpoint", unless = "#result.isEmpty()")
```

Comment:
Shouldn't be Cacheable?

Reply:
Cache is enabled for the trusted certificates now.